File: C++/프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string word) {
    vector<int> orderSum = {781, 156, 31, 6, 1};
    vector<int> wordList = {0, 0, 0, 0, 0};
    vector<char> vowel = {' ', 'A', 'E', 'I', 'O', 'U'};
    
    int answer = 0;
    
    for(int i = 0; i < word.size(); i++) {
        while(vowel[wordList[i]] != word[i]) {
            wordList[i] += 1;
        }
        answer += 1;
        if(1 < wordList[i]) {
            answer += (orderSum[i] * (wordList[i] - 1));
        }
    }
    return answer;
}

File: Python3/프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    order_sum = [781, 156, 31, 6, 1]
    word_list = [0, 0, 0, 0, 0]
    vowel = ['', 'A', 'E', 'I', 'O', 'U']
    answer = 0
    
    
    for i, w in enumerate(word):
        while vowel[word_list[i]] != w:
            word_list[i] += 1
        answer += 1
        if word_list[i] > 1:
            answer += (order_sum[i] * (word_list[i] - 1))
    return answer